### PR TITLE
postgresql@9.5 9.5.22, postgresql@9.6 9.6.18, postgresql@10 10.13,  postgresql@11 11.8, libpqxx 7.1.1

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by alebcay** [comment]
```
clang -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -Wno-unused-command-line-argument -O2  -bundle -multiply_defined suppress -o hstore_plperl.so hstore_plperl.o  -L../../src/port -L../../src/common  -R/usr/local/Cellar/postgresql@10/10.13/lib/postgresql -L/usr/local/opt/openssl@1.1/lib -L/usr/local/opt/readline/lib  -Wl,-dead_strip_dylibs   -bundle_loader ../../src/backend/postgres -fstack-protector  -L/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE -lperl
/usr/bin/install -c -m 755  hstore_plperl.so '/usr/local/Cellar/postgresql@10/10.13/lib/hstore_plperl.so'
install: hstore_plperl.so: No such file or directory
```

Fails on Catalina only. Any ideas on this? This formula in the PR built on my Catalina machine without any issues.

**Comment by SMillerDev** [comment]
The test here times out because it's probably not killed properly.
```
brew test --verbose stellar-core
```

**Comment by alebcay** [comment]
The test actually worked without issues before I rebased/re-pushed earlier, so I wonder if the test might just be too long.

**Comment by Bo98** [comment]
It is a lengthy test. Usually it can be completed in 2 and a half minutes but I guess 10.13 was a little slow on this occasion and took over 5 minutes.

**Comment by SMillerDev** [comment]
Can we reduce the test? (I see it has a bunch of arguments and maybe it can work with less of them?)

**Comment by Bo98** [comment]
Can you incorporate the change in #54985 into the postgresql@11 and postgresql@10 formulae here?

(9.5 and 9.6 still need `XML2_CONFIG`)

**Comment by alebcay** [comment]
Incorporated the above mentioned changes. Maybe the test will just randomly pass this time 🤷.

**Comment by Bo98** [comment]
Not really surprised it works now.

File: Formula/postgresql@10.rb [hunk 1]
```diff
@@ -59,7 +55,17 @@ def install
...
+      inreplace "configure",
+                "-I$perl_archlibexp/CORE",
+                "-iwithsysroot $perl_archlibexp/CORE"
+      inreplace "contrib/hstore_plperl/Makefile",
+                "$(perl_archlibexp)/CORE",
+                "-iwithsysroot $(perl_archlibexp)/CORE"
+      inreplace "src/pl/plperl/GNUmakefile",
+                "$(perl_archlibexp)/CORE",
+                "-iwithsysroot $(perl_archlibexp)/CORE"
+    end
```
Comment (Bo98) [inline Formula/postgresql@10.rb:68]: Has this regressed? It was fixed in 10.6. In fact, it seems like it was backported to 9.5 and 9.6.

On that note, I also suspect the `tclconfig` stuff is also no longer needed under any version after https://github.com/postgres/postgres/commit/5e22171310f8d7c82219a6b978440e5144e88683, but I haven't tested it.

Comment (alebcay) [reply Formula/postgresql@10.rb:68]: The build for `postgresql@10` was broken on something about `hstore_plperl` until I copied the `inreplace`s from the 9.5/9.6 formula into here.

Comment (Bo98) [reply Formula/postgresql@10.rb:68]: Hmm, ok. Bit unusual since `hstore_plperl` hasn't been touched on the 10.x branch since 2018.

Comment (alebcay) [reply Formula/postgresql@10.rb:68]: > On that note, I also suspect the tclconfig stuff is also no longer needed under any version after postgres/postgres@5e22171, but I haven't tested it.

I'll test this locally first and if it works, I'll open a separate PR for it.

Comment (Bo98) [reply Formula/postgresql@10.rb:68]: Yeah, I'll merge this as-is and we can experiment in a separate PR. I'm just nitpicking on cleaning up workarounds that may no longer be needed - they pile up a bit so it's nice when they can be removed (like with the LLVM formula).

**Review by Bo98 — APPROVED** [review]

**Pull request closed without merging**